pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1021,10 +1021,12 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.ClientTLSS
 			tlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs = append(tlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs,
 				authn_model.ConstructSdsSecretConfig(authn_model.SDSDefaultResourceName, opts.push.Mesh.SdsUdsPath))
 
-			tlsContext.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_CombinedValidationContext{
-				CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
-					DefaultValidationContext:         &auth.CertificateValidationContext{MatchSubjectAltNames: util.StringToExactMatch(tls.SubjectAltNames)},
-					ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(authn_model.SDSRootResourceName, opts.push.Mesh.SdsUdsPath),
+			tlsContext.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContext{
+				ValidationContext: &auth.CertificateValidationContext{
+					MatchSubjectAltNames: util.StringToExactMatch(tls.SubjectAltNames),
+					TrustedCa: &core.DataSource{
+						Specifier: &core.DataSource_Filename{Filename: "./var/run/secrets/kubernetes.io/serviceaccount/ca.crt"},
```

Comment:
That will break VMs. 

In general - I like using a filename instead of SDS for root CA, but needs to be customisable.
Or the agent could detect and copy ( or even concat) various sources. 

But we can't hardcode k8s ca.

Reply:
We probably still want to use SDS for the root ca, I was stuck on something while implementing originally and didn't realize I could revert this now that its fixed. I *think* it will "just work" if I undo this. 

Not sure if there are other reasons to use file - but probably is orthogonal to this change and can be done before/after we make this?

Our new VM test failed :slightly_smiling_face: